lib/class-wp-theme-json.php

```diff
@@ -504,10 +493,25 @@ private static function process_key( $key, &$input, $schema, $should_escape = fa
 				if ( 'gradient' === $property ) {
 					$name = 'background';
 				}
-				$result = safecss_filter_attr( "$name: $value" );
 
-				if ( '' === $result ) {
-					unset( $input[ $key ][ $property ] );
+				if ( is_array( $value ) ) {
+					$result = array();
+					foreach ( $value as $subproperty => $subvalue ) {
+						$result_subproperty = safecss_filter_attr( "$name: $subvalue" );
```

Comment:
We are calling safecss_filter_attr with things like:
```
string(21) "background-color: 8px"
string(22) "background-color: 50px"
string(21) "background-color: 8px"
string(21) "background-color: 9px"
```
Possible to test by setting a user padding and adding `var_dump( "$name: $subvalue" );`before safecss_filter_attr filter call.

```
	$name = 'background-color';
	if ( 'gradient' === $property ) {
		$name = 'background';
	}
```


Name is always background-color or background we need to make sure the name also accounts for the padding properties.

Reply:
This is intentional, safecss_filter_attr does not check fine-grained semantics of the declaration, it only has 3 cases: 1) the value is fine (does not contain slashes, parenthesis, etc), 2) the value can contain gradients if the property is allowed to have gradients, and 3) the value can contain URL data if the property is allowed to do so. As per properties, as long they're is in the allowed list, things are fine.

Via theme.json we output more properties than the allowed by the kses function, for example, link color (a CSS Custom Property). I thought the current approach was simpler and equally robust than transforming every property to its CSS form except for link color, which we'd need to transform to other thing.

How do you feel about this?

Reply:
This PR doesn't change this behavior so went ahead and merge it. We can continue to discuss whether this is how we want it to behave.

Reply:
>  I thought the current approach was simpler and equally robust than transforming every property to its CSS form except for link color, which we'd need to transform to another thing.

> How do you feel about this?

Hi @nosolow, I think we need to update and use the correct CSS properties because using just two hardcoded properties is not something expected. E.g: A plugin may change to allow using filter safecss_filter_attr_allow_css a specific CSS only on a specific property, and with this approach, we don't respect what the plugin wants to do.